sentry-android-core/src/main/java/io/sentry/android/core/ANRWatchDog.java

```diff
@@ -73,16 +83,39 @@ public void run() {
       }
 
       // If the main thread has not handled ticker, it is blocked. ANR.
-      if (tick.get() != 0 && !reported) {
-        //noinspection ConstantConditions
+      if (tick.get() != 0 && !reported.get()) {
         if (!reportInDebug && (Debug.isDebuggerConnected() || Debug.waitingForDebugger())) {
           logger.log(
               SentryLevel.DEBUG,
               "An ANR was detected but ignored because the debugger is connected.");
-          reported = true;
+          reported.set(true);
           continue;
         }
 
+        // we only raise an ANR event if the process is in ANR state.
+        // if ActivityManager is not available, we'll still be able to send ANRs
+        final ActivityManager am =
+            (ActivityManager) context.getSystemService(Context.ACTIVITY_SERVICE);
+
+        if (am != null) {
+          final List<ActivityManager.ProcessErrorStateInfo> processesInErrorState =
+              am.getProcessesInErrorState();
+          // if list is null, there's no process in ANR state.
+          if (processesInErrorState == null) {
+            continue;
+          }
+          boolean isAnr = false;
+          for (ActivityManager.ProcessErrorStateInfo item : processesInErrorState) {
+            if (item.condition == NOT_RESPONDING) {
```

Comment:
So this is a best effort since it could give false positive? Meaning there's some process in ANR state but this isn't the current one? Either way would improve **a lot** already.

Reply:
it's only the processes of this app, so that's why it's a list. an App can have multiple processes but its not common at all.